### DISCUSSION TITLE
next build to generate client-side rendering single-page applications, whether or not the getServerSideProps method is used.

### DISCUSSION THREAD

**Discussion by likuner** [vercel/next.js#51776] [Help]
### Summary

Can I use next build to generate client-side rendering single-page applications? Whether or not the getServerSideProps method is used.

**Comment by icyJoseph** [comment]
In Pages Dir, all apps are SPAs, but with a twist, they have an initial frame that's already contentful. 

If you navigate to my landing page from an external source, you'll receive a fully generated HTML document, which was either pre-rendered, or made on demand. 

However, if from the landing page, you navigate to another route, SPA navigation kicks in, and that page is rendered used CSR, with javascript that was generated at build time, and JSON that was either made at build time, or generated on demand.

**Reply by likuner** [reply]
How to build a pure CSR single-page application  that doesn't depend on node services, whether or not I use the getServerSideProps method in my code. any directive, like 'next build'?

**Reply by icyJoseph** [reply]
Use the export feature. Follow these instructions: https://nextjs.org/docs/app/building-your-application/deploying/static-exports

- use output export on next.config.js
- run build

Now you have an `out` directory that contains the application, and only needs a static assets server to work, as any CRA or Vite App do.

**Reply by Dxig** [reply]
And add ```'use client'``` at ```RootLayout.tsx```

**Reply by icyJoseph** [reply]
That makes it both SSR and CSR 😮 https://github.com/reactwg/server-components/discussions/4

I am not really sure about what the OP really asked for here...
